src/main/java/rx/schedulers/ExecutorScheduler.java

```diff
@@ -73,11 +66,15 @@ public Subscription schedule(Action0 action) {
             if (isUnsubscribed()) {
                 return Subscriptions.unsubscribed();
             }
-            ExecutorAction ea = new ExecutorAction(action, tasks);
+            ScheduledAction ea = new ScheduledAction(action, tasks);
```

Comment:
Couldn't we include a reference to `queue` in the `ScheduledAction` so that if it is unsubscribed it can remove itself from the queue?